Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -155,34 +152,20 @@ void FindSXPeaks::exec() {
     double rightBinEdge = *std::next(leftBinPosition);
     double tof = 0.5 * (leftBinEdge + rightBinEdge);
 
-    Geometry::IDetector_const_sptr det;
-    try {
-      det = localworkspace->getDetector(static_cast<size_t>(i));
-    } catch (Mantid::Kernel::Exception::NotFoundError &) {
-      // Catch if no detector. Next line tests whether this happened - test
-      // placed
-      // outside here because Mac Intel compiler doesn't like 'continue' in a
-      // catch
-      // in an openmp block.
-    }
     // If no detector found, skip onto the next spectrum
-    if (!det)
+    if (!spectrumInfo.hasDetectors(static_cast<size_t>(i))) {
       continue;
+    }
+    const auto &det = spectrumInfo.detector(static_cast<size_t>(i));
 
-    double phi = det->getPhi();
+    double phi = det.getPhi();
     if (phi < 0) {
       phi += 2.0 * M_PI;
     }
 
-    double th2 = det->getTwoTheta(sample, L1);
-
     std::vector<int> specs(1, i);
 
-    Mantid::Kernel::V3D L2 = det->getPos();
-    L2 -= sample;
-
-    SXPeak peak(tof, th2, phi, *maxY, specs, l1 + L2.norm(), det->getID(),
-                localworkspace->getInstrument());
```

Comment:
You are correct with the detector group vs. single detector stuff here. For me it seems this is a problem with `Mantid::Geometry::IPeak` which is not designed to work with detector groups. The algorithm outputs a `PeaksWorkspace`, which consists of `IPeak`s which in turn hold information like *Q*, *Intensity* but also detector ID. Now, with my changes, the *Q* and *Intensity* can be for a detector group but the ID is still for a single detector. This could be fixed by getting these values for single detectors, but I suppose there is no way of getting *Intensity* for a single detector in a group, is there? We could also check for `hasUniqueDetector()` here and give warning/bail out but would this change in behavior be undesirable?

With this single detector ID going to the output workspace, I don't think I actually fixed anything here. I could switch to `DetectorInfo`, but then again the intensity data would come from the spectrum itself, if I understood things right. Maybe it would be the right thing to keep as close to the old behavior as possible? Let me know your thoughts about this.